ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/messages/LimitedTransactionsMessagesTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.eth.messages;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import tech.pegasys.pantheon.ethereum.core.BlockDataGenerator;
+import tech.pegasys.pantheon.ethereum.core.Transaction;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Test;
+
+public class LimitedTransactionsMessagesTest {
+
+  private static final int LIMIT = 1048576;
+
+  private final BlockDataGenerator generator = new BlockDataGenerator();
+  private final Set<Transaction> sampleTxs = generator.transactions(1);
+  private final TransactionsMessage sampleTransactionMessages =
+      TransactionsMessage.create(sampleTxs);
+  private final LimitedTransactionsMessages sampleLimitedTransactionsMessages =
+      new LimitedTransactionsMessages(sampleTransactionMessages, sampleTxs);
+
+  @Test
+  public void createLimited() {
+    final Set<Transaction> txs = generator.transactions(6000);
+    final LimitedTransactionsMessages firstMessage = LimitedTransactionsMessages.createLimited(txs);
+    assertEquals(5219, firstMessage.getIncludedTransactions().size());
+    txs.removeAll(firstMessage.getIncludedTransactions());
+    assertEquals(781, txs.size());
+    final LimitedTransactionsMessages secondMessage =
+        LimitedTransactionsMessages.createLimited(txs);
+    assertEquals(781, secondMessage.getIncludedTransactions().size());
+    txs.removeAll(secondMessage.getIncludedTransactions());
+    assertEquals(0, txs.size());
+    assertTrue(
```

Comment:
This test isn't quite right: if you have different inputs (say you generate 2 x 5219 tx's), this check won't necessarily hold.  

The intention here is to check that no single message will ever exceed the max expected size.  I think that would be easier to check with a separate test that explicitly tries to hit the max message size.  You can do that by processing 3 transactions with a size of `LIMIT` and checking that the first 2 are packed up together.  Then process another 3 tx's with sizes {`LIMIT`,` LIMIT` + 1, `LIMIT`} so that each transaction is packaged individually.

So, I would cut this assertion and write a separate test that checks the max message size.